src/worker/worker.c

```diff
@@ -237,6 +240,38 @@ static int finish_job(child_process *cp, int reason)
 }
 
 /*
+ * Get the parent PID from a PID
+ */
+static int get_process_parent_id(const pid_t pid, pid_t * ppid) {
+        char buffer[BUFSIZ], *s_ppid;
+        int errreading;
+        FILE *fp;
+
+        sprintf(buffer, "/proc/%d/stat", pid);
+        fp = fopen(buffer, "r");
```

Comment:
Is it not possible to use `open` or what's the reason for using `fopen`?

Reply:
Yes we can use `open` instead of `fopen`. But I think it's better to use this library function than this system call.

Reply:
Why is it better?

Reply:
I found a nice reason here: https://stackoverflow.com/questions/1658476/c-fopen-vs-open

Reply:
I found a nice reason here: https://stackoverflow.com/questions/1658476/c-fopen-vs-open

Reply:
Thanks, I learned something new :)